src/components/OrderableList/OrderableList.tsx

```diff
@@ -0,0 +1,251 @@
+        },
+    });
+
+    const { collectionProps } = useDroppableCollection(
+        {
+            keyboardDelegate,
+            onDrop: async (event) => {
+                if (event.target.type !== "root" && event.target.dropPosition !== "on" && onMove) {
+                    const keys = [];
+                    for (const item of event.items) {
+                        if (item.kind === "text" && item.types.has(dragTypeId)) {
+                            const key = JSON.parse(await item.getText(dragTypeId));
+                            keys.push(key);
+                        }
+                    }
+
+                    onMove(keys, event.target);
+                }
+            },
+            getDropTargetFromPoint(x, y) {
+                if (gridRef.current) {
+                    const rect = gridRef.current.getBoundingClientRect();
+                    x += rect.x;
+                    y += rect.y;
+                    let closest = null;
+                    let closestDistance = Infinity;
+                    let closestDir = null;
+
+                    for (const child of gridRef.current.children) {
+                        if (!(child as HTMLElement).dataset.key) {
+                            continue;
+                        }
+
+                        const childRect = child.getBoundingClientRect();
+                        const points: [number, number, string][] = [
+                            [childRect.left, childRect.top, "before"],
+                            [childRect.right, childRect.top, "before"],
+                            [childRect.left, childRect.bottom, "after"],
+                            [childRect.right, childRect.bottom, "after"],
+                        ];
+
+                        for (const [px, py, dir] of points) {
+                            const dx = px - x;
+                            const dy = py - y;
+                            const d = dx * dx + dy * dy;
+                            if (d < closestDistance) {
+                                closestDistance = d;
+                                closest = child;
+                                closestDir = dir;
+                            }
+                        }
+
+                        if (y >= childRect.top + 10 && y <= childRect.bottom - 10) {
+                            closestDir = "on";
+                        }
+                    }
+
+                    const key = (closest as HTMLElement)?.dataset.key;
+
+                    if (key) {
+                        return {
+                            type: "item",
+                            key,
+                            dropPosition: closestDir,
+                        } as DropTarget;
+                    }
+                }
+
+                return null;
+            },
+        },
+        dropState,
+        gridRef,
+    );
+
+    //Typeahead must be disabled if list item contains a contenteditable or text input element.
```

Comment:
```suggestion
    // Typeahead must be disabled if list item contains a contenteditable or text input element.
```
Sorry, triggered me as all the others have spaces 😅

Reply:
fixed